modules/gmc-agent/src/main/java/org/gridgain/action/controller/SecurityActionsController.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gridgain.action.controller;
+
+import org.apache.ignite.IgniteAuthenticationException;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.plugin.security.SecurityCredentials;
+import org.gridgain.action.annotation.ActionController;
+import org.gridgain.action.annotation.WithoutAuth;
+import org.gridgain.agent.AgentUtils;
+import org.gridgain.dto.action.AuthenticateCredentials;
+import org.gridgain.action.Session;
+import org.gridgain.action.SessionRegistry;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Controller for security actions.
+ */
+@WithoutAuth
+@ActionController("SecurityActions")
+public class SecurityActionsController {
```

Comment:
May be "SecureActionsController" will be a better name?

Reply:
But all actions in SecureActionsController will be not secure :)  I don't know, the "SecureActionsController" is bad also :(

Reply:
Lest stay with SecurityActionsController